File: app/src/main/java/com/cybersiren/android/v2v/model/VehicleType.kt
package com.cybersiren.android.v2v.model

import android.content.Context
import com.cybersiren.android.R
import com.cybersiren.android.v2v.ui.localized

enum class VehicleType(val code: Int, val displayName: String, val emoji: String) {
    AMBULANCE(1, "Ambulância", "\uD83D\uDE91"),
    FIRE_TRUCK(2, "Bombeiros", "\uD83D\uDE92"),
    POLICE_CAR(3, "Polícia", "\uD83D\uDE93"),
    EMERGENCY(4, "Emergência", "\u26A0\uFE0F");

    fun localizedName(context: Context): String = context.localized(
        when (this) {
            AMBULANCE -> R.string.v2v_vehicle_ambulance
            FIRE_TRUCK -> R.string.v2v_vehicle_fire
            POLICE_CAR -> R.string.v2v_vehicle_police
            EMERGENCY -> R.string.v2v_vehicle_emergency
        }
    )

    companion object {
        fun fromCode(code: Int): VehicleType? = values().find { it.code == code }
    }
}

enum class AlertType(val code: Int, val displayName: String) {
    APPROACHING(1, "Aproximando"),
    PASSING(2, "Passando"),
    STATIONARY(3, "Parado"),
    LEAVING(4, "Afastando");

    companion object {
        fun fromCode(code: Int): AlertType? = values().find { it.code == code }
    }
}

enum class AlertMode {
    SENDER,
    RECEIVER
}

File: ios/bitchat/bitchat/V2V/Models/VehicleType.swift
import Foundation

public enum VehicleType: Int, CaseIterable, Codable {
    case ambulance = 1
    case fireTruck = 2
    case policeCar = 3
    case emergency = 4

    public var displayName: String {
        switch self {
        case .ambulance: return "Ambulância"
        case .fireTruck: return "Bombeiros"
        case .policeCar: return "Polícia"
        case .emergency: return "Emergência"
        }
    }

    public var emoji: String {
        switch self {
        case .ambulance: return "\u{1F691}"
        case .fireTruck: return "\u{1F692}"
        case .policeCar: return "\u{1F693}"
        case .emergency: return "\u{26A0}\u{FE0F}"
        }
    }

    public static func fromCode(_ code: Int) -> VehicleType? {
        return VehicleType(rawValue: code)
    }
}

public enum AlertType: Int, Codable {
    case approaching = 1
    case passing = 2
    case stationary = 3
    case leaving = 4

    public static func fromCode(_ code: Int) -> AlertType? {
        return AlertType(rawValue: code)
    }
}

public enum AlertMode: Codable {
    case sender
    case receiver
}
